plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -64,27 +73,56 @@ struct SoftwareModuleUpdate {
     pub path: Option<String>,
 }
 
+#[derive(Debug, Deserialize)]
+struct TedgeConfig {
+    pub apt: AptConfig,
+}
+
+#[derive(Debug, Deserialize)]
+struct AptConfig {
+    pub name: Option<String>,
+    pub maintainer: Option<String>,
+}
+
 fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
     let status = match operation {
-        PluginOp::List {} => {
-            let apt = Command::new("apt")
-                .args(vec!["--manual-installed", "list"])
+        PluginOp::List { name, maintainer } => {
+            let dpkg_query = Command::new("dpkg-query")
+                .args(vec![
+                    "-f",
+                    "${Package}\t${Version}\t${Maintainer}\t${Status}\n",
+                    "-W",
+                ])
                 .stdout(Stdio::piped())
                 .spawn()
-                .map_err(|err| InternalError::exec_error("apt", err))?
+                .map_err(|err| InternalError::exec_error("dpkg-query", err))?
                 .wait_with_output()
-                .map_err(|err| InternalError::exec_error("apt", err))?;
-
-            let stdout = String::from_utf8(apt.stdout.as_slice().to_vec()).unwrap_or_default();
-            for line in stdout.trim_end().split('\n') {
-                if line == "Listing..." {
-                    // Ignore the first line of the stdout
-                    continue;
+                .map_err(|err| InternalError::exec_error("dpkg-query", err))?;
+
+            let stdout = String::from_utf8(dpkg_query.stdout).unwrap_or_default();
+
+            let filter = match (&name, &maintainer) {
+                (None, None) => Regex::new(r"install ok installed").unwrap(),
+
+                _ => Regex::new(
+                    format!(
+                        r"(^{}\t.*|^\S+\t\S+\t{}\s+.*)install ok installed",
+                        name.unwrap_or_default(),
+                        maintainer.unwrap_or_default()
+                    )
+                    .as_str(),
+                )
+                .unwrap(),
```

Comment:
remove unwrap

Reply:
This unwrap must be replaced by a proper error handler as the regex is built from user inputs: the filters on package names and maintainers.